dagger-core/src/main/java/io/odpf/dagger/core/processors/internal/processor/function/functions/JSONPayloadFunction.java

```diff
@@ -0,0 +1,83 @@
+package io.odpf.dagger.core.processors.internal.processor.function.functions;
+
+import io.odpf.dagger.core.exception.InvalidConfigurationException;
+import io.odpf.dagger.core.processors.internal.processor.function.FunctionProcessor;
+import io.odpf.dagger.common.configuration.Configuration;
+import io.odpf.dagger.core.processors.internal.InternalSourceConfig;
+import io.odpf.dagger.core.processors.common.RowManager;
+import io.odpf.dagger.common.serde.proto.protohandler.TypeInformationFactory;
+import io.odpf.dagger.common.core.StencilClientOrchestrator;
+
+
+import static io.odpf.dagger.common.core.Constants.INPUT_STREAMS;
+import static io.odpf.dagger.common.core.Constants.STREAM_INPUT_SCHEMA_PROTO_CLASS;
+
+import com.google.protobuf.Descriptors;
+import com.google.gson.Gson;
+import org.apache.flink.formats.json.JsonRowSerializationSchema;
+import org.apache.flink.types.Row;
+
+import java.util.Map;
+import java.io.Serializable;
+
+public class JSONPayloadFunction implements FunctionProcessor, Serializable {
+    public static final String JSON_PAYLOAD_FUNCTION_KEY = "JSON_PAYLOAD";
+
+    private InternalSourceConfig internalSourceConfig;
+    private Configuration configuration;
+    private JsonRowSerializationSchema jsonRowSerializationSchema;
+
+    public JSONPayloadFunction(InternalSourceConfig internalSourceConfig, Configuration configuration) {
+        this.internalSourceConfig = internalSourceConfig;
+        this.configuration = configuration;
+        this.jsonRowSerializationSchema = createJsonRowSerializationSchema();
+    }
+
+    @Override
+    public boolean canProcess(String functionName) {
+        return JSON_PAYLOAD_FUNCTION_KEY.equals(functionName);
+    }
+
+    @Override
+    public Object getResult(RowManager rowManager) {
+        return getJSONPayload(rowManager);
+    }
+
+    /**
+     * Gets payload in JSON.
+     *
+     * @return the incoming message as JSON
+     */
+    private String getJSONPayload(RowManager rowManager) {
+        Row inputRow = rowManager.getInputData();
+        return new String(jsonRowSerializationSchema.serialize(inputRow));
+    }
+
+    /**
+     * Creates JSON schema for input row.
+     *
+     * @return the JSON row serialization schema for input row
+     */
+    private JsonRowSerializationSchema createJsonRowSerializationSchema() {
+        if (configuration == null) {
+            throw new InvalidConfigurationException("Invalid configuration: null");
+        }
+
+        String jsonArrayString = configuration.getString(INPUT_STREAMS, "");
+        Gson gsonObj = new Gson();
+        Map[] streamsConfig = gsonObj.fromJson(jsonArrayString, Map[].class);
+        if (streamsConfig == null || streamsConfig.length == 0) {
+            throw new InvalidConfigurationException(String.format("Invalid configuration: %s not provided", INPUT_STREAMS));
+        }
+
+        String inputProtoClassName = (String) streamsConfig[0].get(STREAM_INPUT_SCHEMA_PROTO_CLASS);
```

Comment:
is it only for first stream? 🤔

Reply:
All input streams will have the same proto class right (iirc that's the constraint in our DE console)? Otherwise can expose a single top level proto class config?

Reply:
No, each one can have different proto.

Reply:
I think you should have a different config for this, and not take from Input stream. This is postprocessor config. 
Then you don't need to pass dagger configuration.

Reply:
Had thought about this, but wondering if it'll expose room for errors, e.g, if user specified one proto in input stream and another in postprocessor config, serialization won't work - is that the correct understanding?

Reply:
But yes, I can specify it as a postprocessor config - that might also make the function more generic - user needs to ensure that the output of `FLINK_SQL_QUERY` matches the proto provided in the separate config.

Does the following format work?

```
PROCESSOR_POSTPROCESSOR_CONFIG={"internal_source":[{"output_field":"payload","value":"JSON_PAYLOAD","type":"function", "schema_proto_class": "com.foo.bar.RestaurantMessage"}]
```

Reply:
Seems like you might need an array. One configuration for each stream.